rustls/src/quic.rs

```diff
@@ -763,6 +763,12 @@ impl<'a> KeyBuilder<'a> {
     }
 }
 
+/// A provider for QUIC initial keys.
+pub trait KeyProvider: Debug + Send + Sync {
```

Comment:
I think this comment could use more context for folks less familiar with Quic. Maybe some pointer to why initial keys are handled differently?